src/Fission.re

```diff
@@ -1,87 +1,85 @@
-let baseURL = "http://localhost:1337";
-let env_username = "ca2c70bc13298c5109ee";
-let env_password = "VlBgonAFjZon2wd2VkTR3uc*p-XMd(L_Zf$nFvACpHQShqJ_Hp2Pa";
-
 type cid = string;
-type auth = {
-  username: string,
-  password: string,
+
+let await = promise =>
+  promise
+  |> Js.Promise.then_(response => Js.Promise.resolve(response##data))
+  |> Js.Promise.catch(Js.Promise.resolve);
+
+let octetHeader = Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+
+module type URL = {
+  let baseUrl: string;
+};
+
+module type Auth = {
+  let username: string;
+  let password: string;
 };
 
-let octetHeader =
-  Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+module Simple = (Base: URL) => {
+  let ipfsURL = Base.baseUrl ++ "/ipfs";
+  let cidsURL = ipfsURL ++ "/cids"
+  let url     = cid => ipfsURL ++ cid;
 
```

Comment:
I left out `login` in this version. It's totally possible to do, but requires extra manual typing and recursive modules. More trouble than its worth, since people will likely have logins immediately, or not at all.